repo2docker/buildpacks/guix/__init__.py

```diff
@@ -0,0 +1,74 @@
+"""BuildPack for guix environments"""
+import os
+
+from ..base import BuildPack, BaseImage
+
+
+class GuixBuildPack(BaseImage):
+    """A guix Package Manager BuildPack"""
+
+    def get_path(self):
+        """Return paths to be added to PATH environment variable"""
+        return super().get_path() + ["/home/${NB_USER}/.guix-profile/bin"]
+
+
+    def get_build_scripts(self):
+        """
+        Install guix package manager version 1.3.0.x86_64-linux, using
+        an unmodified installation script found at
+        https://git.savannah.gnu.org/cgit/guix.git/plain/etc/guix-install.sh
+        """
+        return super().get_build_scripts() + [
+            (
+                "root",
+                """
+                yes | BIN_VER=1.3.0.x86_64-linux \
+                bash /home/${NB_USER}/.local/bin/guix-install.bash
+                """,
+            ),
+
+        ]
+
+    def get_build_script_files(self):
+
+        """Copying guix installation script on the image"""
+        return {
+            "guix/guix-install.bash":
+                "/home/${NB_USER}/.local/bin/guix-install.bash",
```

Comment:
If the built image is used with a JupyterHub (as it often is), then everything under $HOME will often instead be replaced with a persistent directory. So ideally buildpacks should leave $HOME clean. Other buildpacks seem to put their libraries under `${APP_BASE}/<something>` (see https://github.com/jupyterhub/repo2docker/blob/de7468d78f6309f6eb7fbd80f2e1758e0efc6b52/repo2docker/buildpacks/conda/__init__.py#L34).

Can we do the same here? Where does everything install to?

Reply:
Done, $HOME should be clear now, let me now what you think!